org.eclipse.xtext/src/org/eclipse/xtext/serializer/impl/Serializer.java

```diff
@@ -185,6 +197,10 @@ public String serialize(EObject obj, SaveOptions options) {
 
 	@Override
 	public void serialize(EObject obj, Writer writer, SaveOptions options) throws IOException {
+		if (obj == null || writer == null || options == null) {
+			throw new IllegalArgumentException(
```

Comment:
This is exactly the place where you do not want to guess from a log file, which argument was eventually the `null` when your code is in production. I belive, this would be better.
```java
if (obj == null) { throw new IAE("obj"); }
if (writer == null ) { throw new IAE("write); }
// others...
```

Same for above.